### PR TITLE
add inplace cos

### PR DESCRIPTION
**Pull request by jonpsy**
@gf712  @karlnapf . Inplace cos method for #5010

### CODE REVIEW

**Comment by gf712** [comment]
If it’s in place why do you have two arguments?

**Comment by jonpsy** [comment]
I think I've used the wrong terminology. This method takes preallocated matrix as input, so it can "act as" a place in.
Edit: I think that's good because if it had only 1 param its definition will clash with the cos which returns container.

File: src/shogun/mathematics/linalg/LinalgNamespace.h [hunk 1]
```diff
@@ -1325,6 +1325,41 @@ namespace shogun
 			return result;
 		}
 
+		/** Performs the operation B = cos(A)
+		 *
+		 * This version returns the result in-place.
```
Comment (karlnapf) [inline src/shogun/mathematics/linalg/LinalgNamespace.h]: remove this line

**Review by karlnapf — COMMENTED** [review]
ehm, where is the code for the cos? This seems to be just the wrapper...

**Comment by jonpsy** [comment]
Sorry, didn't catch that. This is just adding void cos(SGMatrix& a, SGMatrix& result) and similarly for SGVector. The backend for cos, sin and log is already implemented...

**Comment by karlnapf** [comment]
But if it was already implemented, why do we need this additional wrapping code?

**Comment by karlnapf** [comment]
I still dont understand: the backend already had the thing implemented but the interface had not?

**Comment by jonpsy** [comment]
Precisely! LinalgNamespace.h uses ```infer_backend->cos``` all the interfaces. There lacked an interface for the following.
```c++
<template T>
void cos(SGMatrix<T>& A, SGMatrix<T>& result)
```
which I added.

**Comment by jonpsy** [comment]
I've pushed the changes of LinalgNamespace in main PR for CI check just to confirm things.

**Pull request merged**